<center>

#  Work on Data preparation, visualization and Machine learning with Python
## Data Science in Production

<img src='https://www.uniquindio.edu.co/info/uniquindio/media/bloque2.png' width="200" height="250" >

**Students:** <br>
<a href='https://github.com/ingHFrancoO'>
    <img src='https://1000marcas.net/wp-content/uploads/2020/02/logo-GitHub.png' width="90" height="50" >
</a>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 
<a href='https://github.com/1juanfpg'>
    <img src='https://1000marcas.net/wp-content/uploads/2020/02/logo-GitHub.png' width="90" height="50" >
</a>
<br>
Humberto Franco Osorio 
&nbsp;&nbsp;&nbsp;&nbsp;-
&nbsp;&nbsp;&nbsp;&nbsp; Juan Felipe Padilla<br>


**Email:** hfranoo@uqvirtual.edu.co - jfpadillag@uqvirtual.edu.co


**Teacher:** [Jose R. Zapata](https://joserzapata.github.io) <br>
<br>
    <a href='https://joserzapata.github.io'>
        <img src='https://1000marcas.net/wp-content/uploads/2020/02/logo-GitHub.png' width="90" height="50" >
    </a>
    &nbsp;
    <a href='https://twitter.com/joserzapata'>
        <img src='https://upload.wikimedia.org/wikipedia/commons/thumb/4/4f/Twitter-logo.svg/1200px-Twitter-logo.svg.png' width="50" height="40" >
    </a>
    &nbsp;&nbsp;&nbsp;&nbsp;
    <a href='https://www.linkedin.com/in/jose-ricardo-zapata-gonzalez/'>
        <img src='https://cdn-icons-png.flaticon.com/512/174/174857.png' width="50" height="50" >
    </a>
</center>

## Table of Contents
1) Import Libreries
2) Load Dataset
3) Data description and cleaning

    - Variable identification
    - Missing  and  Duplicate values
        - Identify missing values
        - Identify duplicated
        - Dataset cleaning
    - Elimination of features that do not provide information
4) Save intermediated transformed data
5) Partial Results

### Introduction
Analyze the relationship between the characteristics of a cell phone and its selling price, with the aim of classifying the price range in which a cell phone is found.

4 different outputs: <br>
&nbsp;&nbsp;&nbsp; **0** -> Low  <br>
&nbsp;&nbsp;&nbsp; **1** -> Medium   <br>
&nbsp;&nbsp;&nbsp; **2** -> High   <br>
&nbsp;&nbsp;&nbsp; **3** -> Very High   <br>

### Main Objective
> What is the price range of a mobile phone according to its features?

## 1. Import Libreries

In [11]:
import pandas as pd
pd.options.plotting.backend = "plotly"
import numpy as np
import re

## 2. Load Dataset

In [12]:
ds = pd.read_csv("../data/raw/CellPhoneDS.csv")

## 3. Data description and cleaning

### Variable identification

<center>

_**Table 1.** List of features and their descriptions in the initial dataset._

|     | Feature name  | Description and values |  Type| % missing |
|:---:|     :---:     | :---: |:---: |:---: | 
| 1 | Unnamed: 0    | Unique identifier of registered mobile phone. | Numeric |  0
| 2 | index         | Unique mobile phone model identifier. | Nominal | 0
| 3 | battery_power | Total energy a battery can store in one time measured in mAh.     | Numeric | 3.77
| 4 | blue          | Has bluetooth or not. <br>0: NO. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;  1: YES.| Nominal| 4.02
| 5 | clock_speed   | Speed at which microprocessor executes instructions.<br> 31 distinct values. | <font color='yellow'>POR DEFINIR</font> | 3.67
| 6 | dual_sim      | Has dual sim support or not.<br>0: NO. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;  1: YES. | Nominal | 3.54
| 7 | fc            | Front Camera mega pixels. | <font color='yellow'>POR DEFINIR</font>  | 4.00
| 8 | four_g        | Has 4G or not. <br>0: NO. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;  1: YES.| Nominal | 3.96
| 9 | int_memory    | Internal Memory in Gigabytes. | <font color='yellow'>Numeric</font>  | 3.86
| 10 | m_dep         | Mobile Depth in cm. | Numeric  | 3.79
| 11 | mobile_wt     | Weight of mobile phone. | Numeric | 3.67
| 12 | n_cores       | Number of cores of processor. | <font color='yellow'>Numeric</font>  | 3.71
| 13 | pc            | Primary Camera mega pixels. | <font color='yellow'>Numeric</font>  | 3.54
| 14 | px_height     | Pixel Resolution Height. | Numeric | 3.86
| 15 | px_width      | Pixel Resolution Width | Numeric | 3.77
| 16 | ram           | Random Access Memory in Megabytes.| <font color='yellow'>Numeric</font>  | 4.02
| 17 | sc_h          | Screen Height of mobile in cm.| <font color='yellow'>Numeric</font>  | 3.87
| 18 | sc_w          | Screen Width of mobile in cm. | <font color='yellow'>Numeric</font>  | 3.58
| 19 | talk_time     | Longest time that a single battery charge will last when you are charget.  | Numeric | 4.01
| 20 | three_g       | Has 3G or not. <br>0: NO. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;  1: YES.| Nominal | 3.64
| 21 | touch_screen  | Has touch screen or not. <br>0: NO. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;  1: YES.| Nominal | 3.56
| 22 | wifi          | Has wifi or not. <br>0: NO. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;  1: YES.| Nominal | 3.64
| 23 | hgv           | <font color='yellow'>POR DEFINIR</font> | | 3.74
| 24 | werf          | <font color='yellow'>POR DEFINIR</font> | Nominal | 0
| 25 | price_range   | Price range is given by categories.<br> &nbsp;&nbsp;&nbsp; **0** -> Low  <br>&nbsp;&nbsp;&nbsp; **1** -> medium   <br>&nbsp;&nbsp;&nbsp; **2** -> High   <br>&nbsp;&nbsp;&nbsp; **3** -> Very High   <br> | Ordinal | 4.58

</center>

The initial working dataset has 25 columns (feature) and 74790 registered mobile phones.

In [13]:
ds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 74790 entries, 0 to 74789
Data columns (total 25 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Unnamed: 0     74790 non-null  int64 
 1   index          74790 non-null  int64 
 2   talk_time      71990 non-null  object
 3   battery_power  71970 non-null  object
 4   pc             72140 non-null  object
 5   three_g        72070 non-null  object
 6   mobile_wt      72050 non-null  object
 7   px_width       71970 non-null  object
 8   sc_h           71895 non-null  object
 9   sc_w           72115 non-null  object
 10  m_dep          71955 non-null  object
 11  touch_screen   72130 non-null  object
 12  fc             71800 non-null  object
 13  four_g         71825 non-null  object
 14  hgv            71990 non-null  object
 15  price_range    71365 non-null  object
 16  blue           71785 non-null  object
 17  n_cores        72015 non-null  object
 18  wifi           72070 non-n

### Missing  and  Duplicate values

#### A) Identify missing values

In [14]:
ds.sample(10).T

13913            1532             49235  \
Unnamed: 0                13913             1532            49235   
index                       943              876             1155   
talk_time                  13.0              6.0             12.0   
battery_power  nhbgvfrtd 56gyub           1064.0           1414.0   
pc                          2.0             13.0              1.0   
three_g                     0.0              1.0              1.0   
mobile_wt      nhbgvfrtd 56gyub  -948961565145.0            101.0   
px_width                  983.0           1569.0            887.0   
sc_h                       12.0             16.0  -948961565145.0   
sc_w                        8.0             13.0  -948961565145.0   
m_dep                       0.1              0.2              0.5   
touch_screen   nhbgvfrtd 56gyub              0.0              1.0   
fc                          0.0              1.0              0.0   
four_g                      0.0  -948961565145.0              1.0   
hgv                        13.0              6.0             12.0   
price_range                 1.0              0.0              1.0   
blue                        1.0              0.0              1.0   
n_cores                     4.0              6.0  -948961565145.0   
wifi           nhbgvfrtd 56gyub              0.0              0.0   
dual_sim                    0.0              1.0              1.0   
ram                      1808.0            654.0           2290.0   
werf                    9156615          9156615          9156615   
int_memory                  7.0             34.0             49.0   
px_height                 584.0           1175.0  -948961565145.0   
clock_speed                 1.6              2.9              2.2   

                          56622    60866             69901             1128   \
Unnamed: 0                56622    60866             69901              1128   
index                      1155     1241               517              1473   
talk_time      nhbgvfrtd 56gyub     14.0              11.0              15.0   
battery_power            1414.0   1221.0            1059.0             541.0   
pc                          1.0     19.0              18.0               0.0   
three_g                     1.0   ??????               1.0               1.0   
mobile_wt                 101.0    152.0             134.0             127.0   
px_width                  887.0   1899.0            1916.0            1873.0   
sc_h                       13.0   ??????              15.0              19.0   
sc_w                        0.0   ??????              13.0  nhbgvfrtd 56gyub   
m_dep                       0.5   ??????               0.1  nhbgvfrtd 56gyub   
touch_screen                1.0      1.0               1.0  nhbgvfrtd 56gyub   
fc                          0.0      7.0               8.0               0.0   
four_g         nhbgvfrtd 56gyub      1.0               0.0               1.0   
hgv            nhbgvfrtd 56gyub     14.0              11.0              15.0   
price_range                 1.0      2.0               3.0  nhbgvfrtd 56gyub   
blue           nhbgvfrtd 56gyub      0.0               0.0               0.0   
n_cores                     4.0      7.0  nhbgvfrtd 56gyub               4.0   
wifi                        0.0      1.0  nhbgvfrtd 56gyub               0.0   
dual_sim                    1.0      0.0               0.0               0.0   
ram                      2290.0   2638.0            3716.0  nhbgvfrtd 56gyub   
werf                    9156615  9156615           9156615           9156615   
int_memory                 49.0     35.0              61.0              10.0   
px_height                 130.0    765.0            1613.0            1836.0   
clock_speed                 2.2      2.3               2.4               1.0   

                 36266    21680            41268  
Unnamed: 0       36266    21680            41268  
index             2406      448              786

The presence of values such as '??????', 'nhbgvfrtd 56gyub', or '-948961565145.0' is observed, as well as values for some features that are totally atypical considering the description of the feature, is important to find which values had every column.

Get unique values per columns.

In [15]:
print("-----------------------------------------------------------------------------------")
for col in ds:
    print(f"{col} = {ds[col].unique()}")
    print("Number option",f"{len(ds[col].unique())}")
    print("-----------------------------------------------------------------------------------")

-----------------------------------------------------------------------------------
Unnamed: 0 = [    0     1     2 ... 74787 74788 74789]
Number option 74790
-----------------------------------------------------------------------------------
index = [1517  911 1493 ... 1192 1495 1181]
Number option 2493
-----------------------------------------------------------------------------------
talk_time = ['11.0' '9.0' '3.0' '-948961565145.0' '6.0' 'nhbgvfrtd 56gyub' '12.0'
 '8.0' '16.0' '20.0' '13.0' '5285988458456.0' '14.0' '7.0' nan '19.0'
 '10.0' '15.0' '4.0' '18.0' '??????' '17.0' '5.0' '2.0']
Number option 24
-----------------------------------------------------------------------------------
battery_power = ['911.0' '1284.0' '1183.0' ... '527.0' '1542.0' '846.0']
Number option 1099
-----------------------------------------------------------------------------------
pc = ['4.0' '14.0' 'nhbgvfrtd 56gyub' '9.0' '11.0' '20.0' '1.0' '??????' '0.0'
 '16.0' '7.0' '5.0' '5285988458456.0' '10.0' 

From a previous analysis of the dataset it was observed that most of the characteristics data are numerical, so a search for missing values is performed by looking for non-numerical values.

In [16]:
atipic_value=[]
for col in ds:
    for i in range(len(ds)):
        if ds[col].dtype == object:
            line = re.sub(r'[a-zA-z/?/-]', 'nan', str(ds[col][i]))
            if re.search("nan", line):
                atipic_value.append(ds[col][i])
missing_value = np.array(atipic_value)
missing_value = np.unique(missing_value)
print(f'Meaningless values in dataset: {str(missing_value)[1:-2]}')

Meaningless values in dataset: '-948961565145.0' '??????' 'nan' 'nhbgvfrtd 56gyub


Analyzing the presence of these values in the different characteristics of the dataset.

In [17]:
print("-------------------------------------------------------------")
for i in range(len(missing_value)):
    print("Valores iguales a ",missing_value[i])
    frec=ds[ds == missing_value[i]].count()
    for col in ds:
        if frec[col] > 0:
            print(f"{col} = {frec[col]}")
print("-------------------------------------------------------------")

-------------------------------------------------------------
Valores iguales a  -948961565145.0
talk_time = 2495
battery_power = 2495
pc = 2495
three_g = 2495
mobile_wt = 2495
px_width = 2495
sc_h = 2495
sc_w = 2495
m_dep = 2495
touch_screen = 2495
fc = 2495
four_g = 2495
hgv = 2495
price_range = 2495
blue = 2495
n_cores = 2495
wifi = 2495
dual_sim = 2495
ram = 2495
int_memory = 2495
px_height = 2495
clock_speed = 2495
Valores iguales a  ??????
talk_time = 2495
battery_power = 2495
pc = 2495
three_g = 2495
mobile_wt = 2495
px_width = 2495
sc_h = 2495
sc_w = 2495
m_dep = 2495
touch_screen = 2495
fc = 2495
four_g = 2495
hgv = 2495
price_range = 2495
blue = 2495
n_cores = 2495
wifi = 2495
dual_sim = 2495
ram = 2495
int_memory = 2495
px_height = 2495
clock_speed = 2495
Valores iguales a  nan
Valores iguales a  nhbgvfrtd 56gyub
talk_time = 2495
battery_power = 2495
pc = 2495
three_g = 2495
mobile_wt = 2495
px_width = 2495
sc_h = 2495
sc_w = 2495
m_dep = 2495
touch_screen = 2495
fc = 2495
f

There is a low presence of these meaningless vouchers in the different characteristics, therefore it is decided to impute these values as NaN values.

In [18]:
for atipic in missing_value:
    ds.replace(atipic, np.nan, inplace=True)
    
ds.isnull().sum()*100/len(ds)


Unnamed: 0        0.000000
index             0.000000
talk_time        13.751838
battery_power    13.778580
pc               13.551277
three_g          13.644872
mobile_wt        13.671614
px_width         13.778580
sc_h             13.878861
sc_w             13.584704
m_dep            13.798636
touch_screen     13.564648
fc               14.005883
four_g           13.972456
hgv              13.751838
price_range      14.587512
blue             14.025939
n_cores          13.718412
wifi             13.644872
dual_sim         13.544592
ram              14.025939
werf              0.000000
int_memory       13.865490
px_height        13.865490
clock_speed      13.678299
dtype: float64

The percentage of null values in each characteristic increases by approximately 9 points, now there is an average percentage of missing values of 13%, which is a low value considering the size of the dataset, therefore it is decided to eliminate the records with a presence of null values.

#### B) Identify duplicated

Knowing that '*_index_*' is a unique indicator of the mobile phone model, it is assumed that mobile phones with the same model number will have the same characteristics.

With this in mind, we now check whether the model number of the cell phone is registered more than once.

In [19]:
print( 'Duplicated Registers: ' + str( ds["index"].duplicated().sum() ))
print( 'Unique Registers: ' + str( len ( ds["index"].unique() ) ) )

Duplicated Registers: 72297
Unique Registers: 2493


There are 2493 unique values in the '*_index_*', which indicates that the dataset has only 2493 mobile phone models and 72297 duplicate records. Therefore these 72297 duplicate records will be eliminated.

#### C) Dataset cleaning
Taking into account the high number of duplicate records, it is decided to eliminate all the records with NaM values, verifying that the number of unique records (2493) is maintained once these records have been eliminated.

In [20]:
ds_copy = ds.copy() #Save copy of original dataframe
ds_copy.dropna(inplace=True,axis=0)
ds_copy.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 23790 entries, 0 to 74782
Data columns (total 25 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Unnamed: 0     23790 non-null  int64 
 1   index          23790 non-null  int64 
 2   talk_time      23790 non-null  object
 3   battery_power  23790 non-null  object
 4   pc             23790 non-null  object
 5   three_g        23790 non-null  object
 6   mobile_wt      23790 non-null  object
 7   px_width       23790 non-null  object
 8   sc_h           23790 non-null  object
 9   sc_w           23790 non-null  object
 10  m_dep          23790 non-null  object
 11  touch_screen   23790 non-null  object
 12  fc             23790 non-null  object
 13  four_g         23790 non-null  object
 14  hgv            23790 non-null  object
 15  price_range    23790 non-null  object
 16  blue           23790 non-null  object
 17  n_cores        23790 non-null  object
 18  wifi           23790 non-n

By eliminating the records with NaN values we were able to reduce the size of the data set to 23790 records.

In [21]:
ds_copy.sample(10).T

48565    13431            59628    14447    16599  \
Unnamed: 0       48565    13431            59628    14447    16599   
index              526     1986             1564     1668     1193   
talk_time         10.0      8.0  5285988458456.0     14.0     11.0   
battery_power    535.0   1139.0           1273.0    962.0   1130.0   
pc                15.0     11.0             14.0     20.0      8.0   
three_g            1.0      1.0  5285988458456.0      1.0      1.0   
mobile_wt        145.0    161.0            178.0    125.0    101.0   
px_width         939.0    999.0  5285988458456.0   1916.0   1579.0   
sc_h              14.0      9.0  5285988458456.0     16.0     14.0   
sc_w               8.0      4.0              5.0      2.0      5.0   
m_dep              0.5      0.5  5285988458456.0      0.9      0.9   
touch_screen       0.0      0.0              0.0      0.0      1.0   
fc                 8.0      6.0              1.0     14.0      0.0   
four_g             1.0      1.0              0.0      0.0      0.0   
hgv               10.0      8.0  5285988458456.0     14.0     11.0   
price_range        2.0      1.0  5285988458456.0      1.0      0.0   
blue               1.0      1.0              1.0      0.0      0.0   
n_cores            8.0      2.0  5285988458456.0      1.0      1.0   
wifi               0.0      0.0  5285988458456.0      0.0      0.0   
dual_sim           1.0      1.0              0.0      1.0      1.0   
ram             2674.0   1850.0            558.0   1491.0    315.0   
werf           9156615  9156615          9156615  9156615  9156615   
int_memory        54.0     58.0             23.0      2.0     27.0   
px_height        710.0    742.0            485.0   1618.0   1495.0   
clock_speed        0.5      0.9              0.7      1.0      1.8   

                         31804            3617             46959    19121  \
Unnamed: 0               31804             3617            46959    19121   
index                     2071             2013             2138     2213   
talk_time                 20.0  5285988458456.0  5285988458456.0      6.0   
battery_power            534.0            999.0            724.0   1648.0   
pc                         1.0             19.0  5285988458456.0      9.0   
three_g                    1.0  5285988458456.0              1.0      1.0   
mobile_wt                189.0            199.0            187.0    170.0   
px_width                1199.0           1616.0           1754.0   1876.0   
sc_h           5285988458456.0  5285988458456.0              7.0      8.0   
sc_w                      12.0             11.0              6.0      1.0   
m_dep                      1.0              0.2              0.5      0.8   
touch_screen   5285988458456.0              1.0              0.0      1.0   
fc                         0.0             11.0  5285988458456.0      6.0   
four_g                     1.0              1.0  5285988458456.0      1.0   
hgv                       20.0  5285988458456.0  5285988458456.0      6.0   
price_range    5285988458456.0              2.0              2.0      1.0   
blue                       0.0  5285988458456.0              0.0      0.0   
n_cores                    5.0  5285988458456.0              7.0      4.0   
wifi           5285988458456.0              0.0  5285988458456.0      0.0   
dual_sim       5285988458456.0              1.0  5285988458456.0      0.0   
ram                     3926.0  5285988458456.0           2311.0    610.0   
werf                   9156615          9156615          9156615  9156615   
int_memory                16.0             64.0             44.0     64.0   
px_height                393.0           1397.0           1273.0   1153.0   
clock_speed    5285988458456.0              2.9  5285988458456.0      0.7   

                 69757  
Unnamed: 0       69757  
index             2100  
talk_time         15.0  
battery_power    671.0  
pc                17.0  
three_g            1.0  
mobile_wt        1

The presence of value *5285988458456.0* is observed in several fields of the dataset, which corresponds to a meaningless value in the dataset being worked, therefore it is decided to select records that contain this value in one of its characteristics.

In [22]:
ds_copy.replace('5285988458456.0',np.nan,inplace=True)
ds_copy.dropna(inplace=True,axis=0)
ds_copy.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 12130 entries, 0 to 74782
Data columns (total 25 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Unnamed: 0     12130 non-null  int64 
 1   index          12130 non-null  int64 
 2   talk_time      12130 non-null  object
 3   battery_power  12130 non-null  object
 4   pc             12130 non-null  object
 5   three_g        12130 non-null  object
 6   mobile_wt      12130 non-null  object
 7   px_width       12130 non-null  object
 8   sc_h           12130 non-null  object
 9   sc_w           12130 non-null  object
 10  m_dep          12130 non-null  object
 11  touch_screen   12130 non-null  object
 12  fc             12130 non-null  object
 13  four_g         12130 non-null  object
 14  hgv            12130 non-null  object
 15  price_range    12130 non-null  object
 16  blue           12130 non-null  object
 17  n_cores        12130 non-null  object
 18  wifi           12130 non-n

By eliminating the records with **5285988458456.0** values we were able to reduce the size of the data set to 12130 records.


Now, we proceed to delete duplicate records in the dataset

In [23]:
ds_copy.drop_duplicates(subset ="index", keep = 'first', inplace = True)
ds_copy.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2328 entries, 0 to 14950
Data columns (total 25 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Unnamed: 0     2328 non-null   int64 
 1   index          2328 non-null   int64 
 2   talk_time      2328 non-null   object
 3   battery_power  2328 non-null   object
 4   pc             2328 non-null   object
 5   three_g        2328 non-null   object
 6   mobile_wt      2328 non-null   object
 7   px_width       2328 non-null   object
 8   sc_h           2328 non-null   object
 9   sc_w           2328 non-null   object
 10  m_dep          2328 non-null   object
 11  touch_screen   2328 non-null   object
 12  fc             2328 non-null   object
 13  four_g         2328 non-null   object
 14  hgv            2328 non-null   object
 15  price_range    2328 non-null   object
 16  blue           2328 non-null   object
 17  n_cores        2328 non-null   object
 18  wifi           2328 non-nul

In [24]:
print( 'Duplicated Registers: ' + str( ds_copy["index"].duplicated().sum() ))
print( 'Unique Registers: ' + str( len ( ds_copy["index"].unique() ) ) )
print(f"Total missing values in the dataset: {ds_copy.isnull().sum().sum()}")

Duplicated Registers: 0
Unique Registers: 2328
Total missing values in the dataset: 0


### Elimination of features that do not provide information

The columns _Unnamed:0_ and _index_ are eliminated since they are only numerical identifiers for the cell phone models consulted by Pedro Perez.

The _werf_ column is eliminated since it only has a single value in all the records of the dataset.

In [28]:
ds_copy.drop(['Unnamed: 0', 'index', 'werf'], axis=1, inplace=True)


## 4. Save intermediated transformed data

Data without:
- Duplicates
- Missings values
- Non informative data for our taks

In [30]:
# reset data frame index
ds_copy.reset_index(inplace=True, drop=True)
ds_copy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2328 entries, 0 to 2327
Data columns (total 22 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   talk_time      2328 non-null   object
 1   battery_power  2328 non-null   object
 2   pc             2328 non-null   object
 3   three_g        2328 non-null   object
 4   mobile_wt      2328 non-null   object
 5   px_width       2328 non-null   object
 6   sc_h           2328 non-null   object
 7   sc_w           2328 non-null   object
 8   m_dep          2328 non-null   object
 9   touch_screen   2328 non-null   object
 10  fc             2328 non-null   object
 11  four_g         2328 non-null   object
 12  hgv            2328 non-null   object
 13  price_range    2328 non-null   object
 14  blue           2328 non-null   object
 15  n_cores        2328 non-null   object
 16  wifi           2328 non-null   object
 17  dual_sim       2328 non-null   object
 18  ram            2328 non-null

In [31]:
ds_copy.to_parquet('../data/interim/CellPhoneDs_clean.parquet')
#ds.to_csv("../data/interim/CellPhoneDs_clean.csv", index = False)


## 5. Partial Results

### Initial raw data:
74790 rows, 25 Columns , memory usage: 10.9 MB

### After cleaning process
2328 rows , 22 Columns , memory usage: 110.0 KB

***
